src/app/searchbox/searchbox.component.ts

```diff
@@ -0,0 +1,281 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, Input, OnInit} from '@angular/core';
+import { Router } from '@angular/router';
+import { HttpService } from 'src/app/shared/methods/http.service';
+import { environment } from 'src/environments/environment';
+import {NgbDatepickerConfig, NgbDateStruct} from '@ng-bootstrap/ng-bootstrap';
+
+
+@Component({
+  selector: 'app-searchbox',
+  templateUrl: './searchbox.component.html',
+  styleUrls: ['./searchbox.component.css']
+})
+export class SearchboxComponent implements OnInit {
+
+  @Input() fromModel: NgbDateStruct | undefined;
+  @Input() toModel: NgbDateStruct | undefined;
+  @Input() tripRB:string = "One-Way";
+  @Input() nonStopRB: string = "Non-Stop";
+  @Input() numberOfPeople: string = "1 Adult";
+  @Input() adult = 1;
+  @Input() countOfChildren = 0;
+  @Input() fromAirport: string = "";
+  @Input() toAirport: string = "";
+  @Input() fromCalendar:string = "";
+  @Input() disableToCalendar = true;
+
+  //Keeping track of the overlays
+  overlay!: string;
+
+  //variables to hold airportData
+  airportData: any;
+
+  //variables to hold error messages for the user
+  calendarErrorMsg:string = "";
+  returnCalendarErrorMsg: string = "";
+  toAirportErrorMsg:string = "";
+  fromAirportErrorMsg: string = "";
+  adultErrorMsg:string = "";
+
+  //variable for the only disabled input
+  placeHolder:string = "No return date needed";
+  minDate: NgbDateStruct;
+
+  constructor(private httpService: HttpService, private router: Router, private config: NgbDatepickerConfig) { 
+    const current = new Date();
+    this.minDate = {
+      year: current.getFullYear(),
+      month: current.getMonth() + 1,
+      day: current.getDate(),
+    }
+  }
+
+  ngOnInit(): void 
+  {
+    //get airport data from airport service
+
+    let url = environment.airportsEndpoint;
+    this.httpService.get(url).subscribe(
+      (res:any) => 
+      {
+        if (res.length == 0){
+          this.adultErrorMsg = "Utopia currently does not offer any airports we service";
+          return;
+        }
+        let tempDict: {[key:string]: Array<string>} = {};
+        for (let i = 0; i < res.length; i++){
+            let airport = res[i];
+            let servicingArea = airport["servicingArea"]["servicingArea"];
```

Comment:
this is confusing. Good place to use an interface to better understand the structure of the response data and remove magic strings.